pypsa/components/components.py

```diff
@@ -0,0 +1,126 @@
+"""
+Components module.
+
+Contains classes and logic relevant to specific component types in PyPSA.
+Generic functionality is implemented in the abstract module.
+"""
+
+from __future__ import annotations
+
+import logging
+import warnings
+from typing import Any
+
+import pandas as pd
+
+from pypsa.components.abstract import Components
+from pypsa.components.types import ComponentTypeInfo, get_component_type
+from pypsa.definitions.structures import Dict
+
+logger = logging.getLogger(__name__)
+
+
+class GenericComponents(Components):
+    """
+    Generic Components class
+
+    This class is used for components that do not have a specific class implementation.
+    All functionality specific to generic types only is implemented here. Functionality
+    for all components is implemented in the abstract base class.
+
+    .. warning::
+        This class is under ongoing development and will be subject to changes.
+        It is not recommended to use this class outside of PyPSA.
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+class Generators(Components):
+    """
+    Generators Components class
+
+    This class is used for components that are generators. All functionality specific to
+    generators is implemented here. Functionality for all components is implemented in
+    the abstract base class.
+
+    .. warning::
+        This class is under ongoing development and will be subject to changes.
+        It is not recommended to use this class outside of PyPSA.
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+
+CLASS_MAPPING = {
+    "Generator": Generators,
+}
```

Comment:
Am I correct that this would have to be expanded by the other components before merging?